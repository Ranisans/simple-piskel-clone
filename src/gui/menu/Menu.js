import React from 'react';

import Label from './Label';
import ClearButton from './ClearButton';
import SignInButton from './SigtInButton';

const Menu = () => (
  <section className="main_menu">
    <h2 className="section_label">Main Menu</h2>
    <h3 className="main_menu-label">Simple Piskel Clone</h3>
    <Label />
    <ClearButton />
    <SignInButton />
  </section>
);

export default Menu;
