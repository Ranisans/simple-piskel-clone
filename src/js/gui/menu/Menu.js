import React from 'react';

import Button from './Button';

const Menu = () => {
  const buttons = [
    { className: 'create_sprite', value: 'Create Sprite' },
    { className: 'login_btn', value: 'SignIn' },
  ];
  return (
    <section className="main_menu">
      <h2 className="section_label">Main Menu</h2>
      <h3 className="main_menu-label">Simple Piskel Clone</h3>
      {buttons.map((el, i) => (
        <Button
          key={i}
          className={el.className}
          value={el.value}
        />
      ))}
    </section>
  );
};

export default Menu;
