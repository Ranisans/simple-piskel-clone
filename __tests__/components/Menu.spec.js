import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Menu from '../../src/gui/menu/Menu';


describe('Menu tests', () => {
  it('create snapshot', () => {
    const component = shallow(
      <Menu />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
