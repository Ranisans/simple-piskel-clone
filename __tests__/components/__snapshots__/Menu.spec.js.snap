// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu tests create snapshot 1`] = `
<section
  className="main_menu"
>
  <h2
    className="section_label"
  >
    Main Menu
  </h2>
  <h3
    className="main_menu-label"
  >
    Simple Piskel Clone
  </h3>
  <Label />
  <ClearButton />
  <SignInButton />
</section>
`;
